File: Lab 4/C#/Figure.cs
namespace FigureLibrary
{
    public class Figure
    {
        protected double x;
        protected double y;
        protected double side;

        // Конструктор за замовчуванням
        public Figure()
        {
            x = 0;
            y = 0;
            side = 1;
        }

        // Конструктор із 3 параметрами
        public Figure(double xVal, double yVal, double sideVal)
        {
            x = xVal;
            y = yVal;
            side = sideVal;
        }

        // Гетери
        public double GetX() { return x; }
public double GetY() { return y; }
public double GetSide() { return side; }

// Метод повертає довжину сторони
public double Length()
{
    return side;
}
    }
}

File: Lab 4/C++/Figure.cpp
#include "Figure.h"

Figure::Figure() : x(0), y(0), side(1) {
    // За замовчуванням: вершина (0,0), сторона = 1
}

Figure::Figure(double xVal, double yVal, double sideVal)
    : x(xVal), y(yVal), side(sideVal) {
}

Figure::Figure(const Figure& other)
    : x(other.x), y(other.y), side(other.side) {
}

double Figure::getX() const {
    return x;
}

double Figure::getY() const {
    return y;
}

double Figure::getSide() const {
    return side;
}

double Figure::length() const {
    // Довжина сторони
    return side;
}